diff --git a/src/meson.build b/src/meson.build
index 00c735c..c5b252e 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -81,7 +81,7 @@ foreach name : clapper_possible_enhancers
           custom_target(enhancer_plugin_template,
             input: join_paths(name, enhancer_plugin_template),
             output: enhancer_plugin_template.replace('.in', ''),
-            command: [py, files('gen-version.py'), '@INPUT@', meson.project_version(), get_option('enhancersver')],
+            command: [py, files('gen-version.py'), '@INPUT@', meson.project_version(), meson.project_version()],
             capture: true,
             build_by_default: true,
             build_always_stale: true,
