diff --git a/gtuber/meson.build b/gtuber/meson.build
index 82721c0..e4297b3 100644
--- a/gtuber/meson.build
+++ b/gtuber/meson.build
@@ -19,18 +19,11 @@ gmodule_dep = dependency('gmodule-2.0',
   fallback: ['glib', 'libgmodule_dep'],
 )
 
-soup_dep_name = 'libsoup-3.0'
-soup_gir_name = 'Soup-3.0'
+soup_dep_name = 'libsoup-2.4'
+soup_gir_name = 'Soup-2.4'
 soup_dep = dependency(soup_dep_name,
-  required: false,
+  required: true,
 )
-if not soup_dep.found()
-  soup_dep_name = 'libsoup-2.4'
-  soup_gir_name = 'Soup-2.4'
-  soup_dep = dependency(soup_dep_name,
-    required: true,
-  )
-endif
 
 version_conf = configuration_data()
 
