diff --git a/sys/va/plugin.c b/sys/va/plugin.c
index d8f5cbf3d..313f46458 100644
--- a/sys/va/plugin.c
+++ b/sys/va/plugin.c
@@ -96,35 +96,35 @@ plugin_register_decoders (GstPlugin * plugin, GstVaDevice * device,
     switch (codec) {
       case H264:
         if (!gst_va_h264_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register H264 decoder: %s",
               device->render_device_path);
         }
         break;
       case HEVC:
         if (!gst_va_h265_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register H265 decoder: %s",
               device->render_device_path);
         }
         break;
       case VP8:
         if (!gst_va_vp8_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register VP8 decoder: %s",
               device->render_device_path);
         }
         break;
       case VP9:
         if (!gst_va_vp9_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register VP9 decoder: %s",
               device->render_device_path);
         }
         break;
       case MPEG2:
         if (!gst_va_mpeg2_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register Mpeg2 decoder: %s",
               device->render_device_path);
         }
@@ -132,7 +132,7 @@ plugin_register_decoders (GstPlugin * plugin, GstVaDevice * device,
 #if VA_CHECK_VERSION(1, 8, 0)
       case AV1:
         if (!gst_va_av1_dec_register (plugin, device, sinkcaps, srccaps,
-                GST_RANK_NONE)) {
+                GST_RANK_PRIMARY + 3)) {
           GST_WARNING ("Failed to register AV1 decoder: %s",
               device->render_device_path);
         }
